### PR TITLE
NIFI-9861: Removed BlockListClassLoader in favor of AllowListClassLoader

### PR DESCRIPTION
**Pull request by markap14**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-stateless/nifi-stateless-bootstrap/src/main/java/org/apache/nifi/stateless/bootstrap/StatelessBootstrap.java [hunk 1]
```diff
@@ -181,58 +166,75 @@ private static BlockListClassLoader createExtensionRootClassLoader(final File na
...
-        // If the classLoader is the system class loader, we are done. We don't want to process the parent of
-        // the system class loader (which would be the Launcher$ExtClassLoader that contains the JDK/JRE classes, etc)
-        if (classLoader != ClassLoader.getSystemClassLoader()) {
-            findClassLoaderUrls(classLoader.getParent(), urls);
+    private static void findLoadableClasses(final File file, final Set<String> classNames) throws IOException {
+        final String filename = file.getName();
+        if (filename.endsWith(".jar")) {
+            findClassNamesInJar(file, classNames);
+        } else if (filename.endsWith(".jmod")) {
+            findClassesInJmod(file, classNames);
```
Comment (turcsanyip) [inline nifi-stateless/nifi-stateless-bootstrap/src/main/java/org/apache/nifi/stateless/bootstrap/StatelessBootstrap.java:237]: `.class` files are collected in `findLoadableFiles()` (line 227) but they are not handled here. It did not cause any issues in my flows but there can be other use cases when those classes are needed.

**Review by turcsanyip — CHANGES_REQUESTED** [review]
@markap14 Thanks for fixing this issue!
I tested it with a simple flow in different environments (run test from IntelliJ, Stateless NiFi, Kafka Connect, ExecuteStateless, all with java 8 and 11). It worked for me.

However, I found a possible issue regarding handling of `.class` files. Pls. see my comment inline.

**Comment by markap14** [comment]
Thanks @turcsanyip . That's a good catch. We can't really add the classes found in .class files because we don't know what the fully qualified class name is. We do know if it's found in a .jar or a .jmod file. Good news is that it doesn't really matter, as it's only used to detect classes that are part of Java itself, and Java packages all of their classes uses the appropriate packing mechanisms (.jmod / .jar) - at least in every case I've found. Hard to imagine that they wouldn't.

So I removed the .class from the list of things we add to the 'loadable files'. I think this is fine and addresses the mismatch between the two. I also rebased as there was a conflict with main.

**Review by turcsanyip — APPROVED** [review]
@markap14 That makes sense. +1 LGTM

Merging to main.

**Pull request closed without merging**